target/linux/mvebu/config-4.9

```diff
@@ -60,6 +60,8 @@ CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_SD=y
 CONFIG_BLK_MQ_PCI=y
 CONFIG_BOUNCE=y
+CONFIG_BTRFS_FS=y
```

Comment:
There is no automated tftp recovery mode. Serial is required to access u-boot for tftp and also as it turns out to use the built-in OpenWrt recovery image.

https://www.turris.cz/doc/en/howto/omnia_factory_reset

The only way I can think to get the Omnia to do anything other than boot from internal btrfs without opening it/serial is to use fw_setenv from the factory OS to set tftp for next boot.

(thinking out loud) It may also be possible to write a script to run on Turris OS that prepares fw_setenv for booting a ClearFog-like image, downloads the LEDE eMMC image to /tmp/ and then dd's it on top of the mounted btrfs filesystem. A dirty hack, but it would avoid having to tftp from another machine.

I quite liked the concept of using the Omnia's built-in recovery and rollback built on btrfs, but writing sysupgrade scripts (and probably also btrfs userland tools) is increasingly seeming like it will be a nuisance all while adding bloat to the mvebu kernel for the sake of bending to one device's factory partition layout.

I will experiment over the coming days with tftp booting my Omnia, flashing FAT/squashfs image using ClearFog sysupgrade scripts and then u-booting from this, and test flashing from a running btrfs environment.